cmake/deps/Lz4.cmake

```diff
@@ -0,0 +1,75 @@
+#[===================================================================[
+   NIH dep: lz4
+#]===================================================================]
+
+add_library (lz4_lib STATIC IMPORTED GLOBAL)
+find_library(lz4 NAMES lz4)
```

Comment:
Two main things are lacking with the simple `find_library` approach: (1) version requirement and (2) static/dynamic use. This currently is broken for me on macos, for example, because find_library finds the dylib even when I do static builds. Furthermore, we've declared a STATIC imported library, but then this searches for any lib which usually defaults to dynamic first. That's inconsistent.

lz4, like most projects, is not producing proper cmake configs. But it looks like it's instead installing package config modules, so I think we need to grab a find module that does the pkgconfig search/imports. https://github.com/scylladb/seastar/blob/master/cmake/Findlz4.cmake might be a good start.  That would give us a version check (I would make 1.8 minimum version because I'm not willing to test all the prior versions) and PkgConfig should provide static and dynamic vars when found. If the the case of static rippled builds, we need to make sure the static libs is being used here.